components/aero_sensors/SConscript

```diff
@@ -0,0 +1,279 @@
+    # "-Wl,--print-gc-sections",
+    "--specs=nano.specs",
+    "--specs=nosys.specs",
+]
+
+as_flags = ["-xassembler-with-cpp"]
+
+c_flags = [
+    "-std=c11",
+    "-ffunction-sections",
+    "-fdata-sections",
+    "-fshort-enums",
+    "-funsigned-char",
+    "-fstack-usage",
+    "-nostdlib",
+    "-msoft-float",
+    "-DSTM32F1",
+    "-DSTM32F103xB",
+    "-DUSE_HAL_DRIVER",
+    "-Wall",
+    "-Wextra",
+    "-Werror",
+    "-Wfloat-equal",
+    "-Wcast-align",
+    "-Wlogical-op",
+    "-Winline",
+    "-Wshadow",
+    "-Winit-self",
+    "-Wmissing-prototypes",
+    "-Wunused-function",
+    "-Wpointer-arith",
+    "-Wno-type-limits",
+    "-Wno-unused-local-typedefs",
+]
+
+free_rtos_flags = [
+    "-Wno-missing-prototypes",
+    "-Wno-cast-align",
+]
+
+chip_config = env.ChipConfig(config_file="HW/mcuConfig.yaml")
+if "defines" in chip_config:
+    c_flags.append(chip_config["defines"])
+
+if GetOption("release"):
+    common_flags += release_flags
+else:
+    common_flags += debug_flags
+
+
+env["LINKFLAGS"] = [link_flags + common_flags]
+
+RTOS_INCLUDE_DIRS = [
+        RTOS_DIR.Dir("include"),
+        FREERTOS_DIR.Dir("Source/include/"),
+        FREERTOS_DIR.Dir("Source/portable/GCC/ARM_CM3/"),
+        CMSIS_DIR.Dir("CMSIS/Core/Include/"),
+]
+
+FS_INCLUDE_DIRS = [
+    FS_DIR.Dir("include"),
+    FATFS_DIR, # Necessary so that integer.h comes from FS/include
+    LIBS_DIR.Dir("FreeRTOS/CMSIS/RTOS2/FreeRTOS/Include1/"),
+    CMSIS_DIR.Dir("CMSIS/RTOS2/Include/"),
+]
+
+DRIVERS_INCLUDE_DIRS = [
+    chip_config["cmsis_includes"],
+    chip_config["hal_includes"],
+]
+
+LIBS_INCLUDE_DIRS = [
+]
+
+PROJECT_INCLUDE_DIRS = [
+    "include/",
+    "HW/include/",
+    "generated/",
+    #"lib/lib_FTDI_FT81x/include",
+]
+
+paths = (
+    FS_INCLUDE_DIRS + RTOS_INCLUDE_DIRS + DRIVERS_INCLUDE_DIRS + PROJECT_INCLUDE_DIRS + LIBS_INCLUDE_DIRS
+)
+
+env.Append(
+    ASFLAGS=as_flags,
+    CPPPATH=paths,
+    CCFLAGS=common_flags + c_flags,
+    LINKSCRIPT=chip_config["linker_file"],
+)
+
+rtos_env.Append(
+    CPPPATH=paths,
+    CCFLAGS=common_flags + c_flags + free_rtos_flags,
+)
+
+# List of component C files
+
+project_source_files = [
+    PROJECT_SOURCE_DIR.File("SystemManager.c"),
+    PROJECT_SOURCE_DIR.File("Module.c"),
+    PROJECT_SOURCE_DIR.File("Sensors.c"),
+    PROJECT_SOURCE_DIR.File("Files.c"),
+    PROJECT_SOURCE_DIR.File("IO.c"),
+]
+
+project_hw_files = [
+        PROJECT_HW_DIR.File("HW_adc.c"),
+        #PROJECT_HW_DIR.File("HW_can.c"),
+        PROJECT_HW_DIR.File("HW_dma.c"),
+        PROJECT_HW_DIR.File("HW_gpio.c"),
+        PROJECT_HW_DIR.File("HW_intc.c"),
+        PROJECT_HW_DIR.File("HW_msp.c"),
+        PROJECT_HW_DIR.File("HW_spi.c"),
+        PROJECT_HW_DIR.File("HW_timebase.c"),
+        PROJECT_HW_DIR.File("HW_i2c.c"),
+        PROJECT_HW_DIR.File("HW_MPRL.c"),
+        PROJECT_HW_DIR.File("HW_MAX7356.c"),
+        PROJECT_HW_DIR.File("HW_NPA.c"),
```

Comment:
these should probably end up in the libs dir rather than being specific to this project. Doesn't need to happen now though